src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -2356,21 +2356,53 @@ def nunique(self, axis=0, dropna=True):  # noqa: PR01, RT01, D200
             result.name = None
         return result
 
-    @base_not_implemented()
     def pct_change(
-        self, periods=1, fill_method="pad", limit=None, freq=None, **kwargs
+        self, periods=1, fill_method=no_default, limit=no_default, freq=None, **kwargs
     ):  # noqa: PR01, RT01, D200
         """
         Percentage change between the current and a prior element.
         """
-        # TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset
-        return self._default_to_pandas(
-            "pct_change",
-            periods=periods,
-            fill_method=fill_method,
-            limit=limit,
-            freq=freq,
-            **kwargs,
+        if fill_method not in (lib.no_default, None) or limit is not lib.no_default:
+            warnings.warn(
+                "The 'fill_method' keyword being not None and the 'limit' keyword in "
+                + f"{type(self).__name__}.pct_change are deprecated and will be removed "
+                + "in a future version. Either fill in any non-leading NA values prior "
+                + "to calling pct_change or specify 'fill_method=None' to not fill NA "
+                + "values.",
+                FutureWarning,
+                stacklevel=1,
+            )
+        if fill_method is lib.no_default:
+            warnings.warn(
+                f"The default fill_method='pad' in {type(self).__name__}.pct_change is "
+                + "deprecated and will be removed in a future version. Either fill in any "
+                + "non-leading NA values prior to calling pct_change or specify 'fill_method=None' "
+                + "to not fill NA values.",
+                FutureWarning,
+                stacklevel=1,
+            )
+            fill_method = "pad"
+
+        if limit is lib.no_default:
+            limit = None
+
+        # Attempting to match pandas error behavior here
+        if not isinstance(periods, int):
+            raise TypeError(f"periods must be an int. got {type(periods)} instead")
+
+        # Attempting to match pandas error behavior here
+        for dtype in self._get_dtypes():
+            if not is_numeric_dtype(dtype):
+                raise TypeError(f"unsupported operand type for /: got {dtype}")
```

Comment:
I'm not sure how to read this error message. Also, can we have a test that exercises it?

Reply:
pandas raises an error message like this because it doesn't do type validation and attempts to do division (`/`) between two non-numeric values. I've now changed the message to something more readable, and added a test for it in `frame/test_pct_change.py`.